zowe/zowe-install-packaging
# Record ZIS params in instance

This PR is a simplification of a part of https://github.com/zowe/zowe-install-packaging/pull/1756 which only deals with how to get ZIS parameters into instance.env

These things are known at install time, but could be customized post-install.
Therefore, though we can automate getting the data into instance.env by recording install output into a temp file, and reading it back at instance creation time.
However, because these params can be customized, automation not be desired.
So, there are arguments to specify them explicitly.

Also, the instance creation script argument parsing was reworked to allow for more flexibility in input. The behavior is backwards compat.

How to test:
If you use zowe as you normally did, you should end up with LOADLIB and PARMLIB ZIS parameters in instance.env when you create a new instance. This doesnt update old instances.
However, if for some reason you customize or things go wrong with the temp file logic, you can explicitly specify loadlib, parmlib, or just the dsn prefix by the -l, -p, and -d flags to the configure instance script.

For example

Failure or what is seen on an instance upgrade:
```
ZWES_ZIS_LOADLIB=
ZWES_ZIS_PARMLIB=
ZWES_ZIS_PARMLIB_MEMBER=ZWESIP00
```

Success when no parameters are given and automation works
```
ZWES_ZIS_LOADLIB=myprefix.SZWEAUTH
ZWES_ZIS_PARMLIB=myprefix.SAMPLIB
ZWES_ZIS_PARMLIB_MEMBER=ZWESIP00
```

Success when parameters were specified
`/bin/zowe-configure-instance.sh -c inst/zis10 -l abc -p def`

```
ZWES_ZIS_LOADLIB=abc
ZWES_ZIS_PARMLIB=def
ZWES_ZIS_PARMLIB_MEMBER=ZWESIP00
```
